java/build/bazel/tests/integration/WorkspaceDriver.java

```diff
@@ -155,6 +151,27 @@ protected void copyFromRunfiles(String path, String destpath) throws IOException
     Files.copy(origin.toPath(), dest.toPath());
   }
 
+  /**
+   * Copy the whole directory from the runfiles under {@code rootDirectoryPath} to the current workspace.
+   */
+  protected void copyFromRunfilesDirectory(final String rootDirectoryPath) throws IOException {
+    File root = getRunfile(rootDirectoryPath);
+    try (Stream<Path> paths = Files.walk(Paths.get(root.toURI()))) {
+      paths.filter(path -> new File(path.toUri()).isFile())
```

Comment:
why do you need to filter out directories? what happens if you don't? this probably makes sense if you're going to forbit it in the "regular" copyFromRunfiles method

Reply:
Because `copyFromRunfiles` shouldn't work for directories, we'll decide how to handle it later.

Also, `walk` do the traversal depth first so I'll get to the directories after the leafs meaning that the directory already exists.